Refined README.md for App-toolkit (#3705)
vmware-tanzu/community-edition: addons/packages/app-toolkit/0.1.0/README.md

> ### Prerequisites

Reviewer: We should add a line to installl ytt as the pre-requisite for now

> * TCE has already been installed and the Tanzu CLI is available.

Reviewer 2: ```suggestion
* The Tanzu CLI is installed, see [Getting Started](https://tanzucommunityedition.io/docs/v0.10/getting-started/#install-tanzu-cli).
```

> * The following steps require the Tanzu CLI app plugin. See [here](https://github.com/vmware-tanzu/apps-cli-plugin#getting-started) on how to install.

Reviewer 2: ```suggestion

```

Reviewer 2: I misread, my mistake, agree, marking resolved @krisapplegate

> * You will need to provide a container registry for both `kpack` and the sample supply chain to leverage. In the examples below a free DockerHub account is shown, but you can reference the [TCE kpack documentation](https://tanzucommunityedition.io/docs/v0.10/package-readme-kpack-0.5.0/) to see examples utilizing other registries.

Reviewer 2: ```suggestion
* You will need to provide a container registry for both `kpack` and the sample supply chain to leverage. In the examples below a free DockerHub account is shown, but you can reference the [Tanzu Community Edition kpack documentation](https://tanzucommunityedition.io/docs/v0.10/package-readme-kpack-0.5.0/) to see examples utilizing other registries.
```

### Prepare a values.yaml file for local Docker installation without load balancer

> In this example we will deploy our traffic ingress mechanism, contour, with a ClusterIP configuration, and Knative to serve at localhost. You'll want to add these settings to a `values.yaml` file. As referenced in the Prerequisites, you'll also need to add a Docker registry for kpack to store buildpacks on. The example below is one method of configuring App-toolkit, feel free to adapt the configuration to reflect your own environment.

Reviewer 2: ```suggestion
In the following example, you will deploy our traffic ingress mechanism, Contour, with a ClusterIP configuration, and Knative to serve as localhost. You will add these settings to a `values.yaml` file, and add a Docker registry for kpack to store buildpacks on. This example is one method of configuring App-toolkit, feel free to adapt the configuration to reflect your own environment.
```

```yaml
contour:
  envoy:
    service:
      type: ClusterIP
    hostPorts:
      enable: true

knative_serving:
  domain:
    type: real
    name: 127-0-0-1.sslip.io

kpack:
  kp_default_repository: https://index.docker.io/v1/
  kp_default_repository_username: [your username]
  kp_default_repository_password: [your password]
```

> ### Installing using the custom values file

Reviewer 2: ```suggestion
### Install App-toolkit using the custom values file
```

  - cert-manager.community.tanzu.vmware.com
```

## Usage Example

This example illustrates creating a simple Spring Boot web app on the locally deployed TCE cluster using Docker.

> 1. Begin with Docker and the Tanzu CLI installed on your development machine and ensure you have the Carvel tools installed as well as we will be leveragig `ytt` in the examples below. For best performance, configure Docker with at least 8GB of RAM and that you have plenty of free space on the local filesystem that Docker is using.

Reviewer: @krisapplegate why do we need 8GB of RAM? Have we validated that?

Author: It was in the previous doc, I just ported it forward.

Reviewer: Let's remove that please. We don't know if that's true.

  - cert-manager.community.tanzu.vmware.com
```

## Usage Example

> This example illustrates creating a simple Spring Boot web app on the locally deployed TCE cluster using Docker.

Reviewer 2: ```suggestion
This example illustrates creating a simple Spring Boot web app on the locally deployed Tanzu Community Edition cluster using Docker.
```

> 1. Begin with Docker and the Tanzu CLI installed on your development machine and ensure you have the Carvel tools installed as well as we will be leveragig `ytt` in the examples below. For best performance, configure Docker with at least 8GB of RAM and that you have plenty of free space on the local filesystem that Docker is using.

Reviewer 2: ```suggestion
1. Begin with Docker and the Tanzu CLI installed on your development machine and ensure you have the Carvel tools installed as well as we will be leveraging `ytt` in the examples below. For best performance, configure Docker with at least 8GB of RAM and ensure that you have plenty of free space on the local filesystem that Docker is using.
```

> 1. Create a cluster named demo-local exposing 80 and 443 ports.

Reviewer 2: ```suggestion
1. Create an unmanaged cluster named demo-local exposing 80 and 443 ports.
```

    ```shell
    tanzu unmanaged-cluster create demo-local -p 80:80 -p 443:443
    ```

> 1. Follow the steps listed in the above "Installation" section to ensure you have a completed installation of the App-toolkit package. You can validate this by checking that all the packages have successfully reconciled using the command

Reviewer 2: ```suggestion
1. Ensure you have installed the App-toolkit package. You can validate this by checking that all the packages have successfully reconciled using the command:
```

      cartographer              cartographer.community.tanzu.vmware.com              0.2.2                  Reconcile succeeded  tanzu-package-repo-global
      cert-manager              cert-manager.community.tanzu.vmware.com              1.6.1                  Reconcile succeeded  tanzu-package-repo-global
      contour                   contour.community.tanzu.vmware.com                   1.19.1                 Reconcile succeeded  tanzu-package-repo-global
      fluxcd-source-controller  fluxcd-source-controller.community.tanzu.vmware.com  0.21.2                 Reconcile succeeded  tanzu-package-repo-global
      knative-serving           knative-serving.community.tanzu.vmware.com           1.0.0                  Reconcile succeeded  tanzu-package-repo-global
      kpack                     kpack.community.tanzu.vmware.com                     0.5.1                  Reconcile succeeded  tanzu-package-repo-global
      cni                       antrea.tanzu.vmware.com                              0.13.3+vmware.1-tkg.1  Reconcile succeeded  tkg-system
      ```

> 1. You'll now need to prepare a second values file (apart from the one used to install App-toolkit) with the additional configuration necessary for `kpack` and `cartographer`. In the below examples, this second values file will be refereced as `supplychain-example-values.yaml`.

Reviewer 2: ```suggestion
1. You'll now need to prepare a second values file (apart from the one used to install App-toolkit) with the additional configuration necessary for `kpack` and `cartographer`. In the below examples, this second values file will be referenced as `supplychain-example-values.yaml`.
```

    ```yaml
    kpack:
      registry:
        url: [REGISTRY_URL]
        username: [REGISTRY_USERNAME]
        password: [REGISTRY_PASSWORD]
      builder:
        # path to the container repository where kpack build artifacts are stored
        tag: [REGISTRY_TAG]
      # A comma-separated list of languages e.g. [java,nodejs] that will be supported for development
      # Allowed values are:
      # - java
      # - nodejs
      # - dotnet-core
      # - go
      # - ruby
      # - php
      languages: [java]
      image_prefix: [REGISTRY_PREFIX]

    ```

    Where:
    * `REGISTRY_URL` is a valid OCI registry to store kpack images. For DockerHub this would be `https://index.docker.io/v1/`
    * `REGISTRY_USERNAME` and `REGISTRY_PASSWORD` are the credentials for the specified registry.
    * `REGISTRY_TAG` is the path to the container repository where kpack build artifacts are stored. For DockerHub, it is username/tag, e.g. csamp/builder

>     * `REGISTRY_PREFIX` is prefix for your images as they reside on the registry. For DockerHub, it is the username followed by a trailing slash, e.g. csamp/

Reviewer 2: ```suggestion
    * `REGISTRY_PREFIX` is the prefix for your images as they reside on the registry. For DockerHub, it is the username followed by a trailing slash, e.g. csamp/
```
